src/main/java/com/ibm/cloud/sdk/core/security/IamToken.java

```diff
@@ -51,28 +52,51 @@ public Long getExpiration() {
     return expiration;
   }
 
+  private IamToken() {
+    if (getExpiresIn() != null && getExpiration() != null) {
+      Double fractionOfTimeToLive = 0.8;
+      Long timeToLive = getExpiresIn();
+      Long expirationTime = getExpiration();
+      this.refreshTime = expirationTime - (long) (timeToLive * (1.0 - fractionOfTimeToLive));
+    }
+  }
+
```

Comment:
It turns out that this constructor is called from GSON as part of the processing the response from the IAM token server.  But the way GSON works is that it will construct an "empty object", and then fill in the fields.  So when this constructor runs, all the fields are `null`, so `this.refreshTime` is never set.

I think what you should delete this constructor and put the logic for calculating `refreshTime` into `needsRefresh` below.